crates/yttrium/src/pay/mod.rs

```diff
@@ -554,24 +523,55 @@ impl WalletConnectPay {
 
 // Private methods (not exported via uniffi)
 impl WalletConnectPay {
+    async fn resolve_actions(
+        &self,
+        payment_id: &str,
+        option_id: &str,
+        actions: Vec<types::RequiredAction>,
+    ) -> Result<Vec<RequiredAction>, GetPaymentRequestError> {
+        let mut result = Vec::new();
+        for action in actions {
+            match action {
+                types::RequiredAction::WalletRpc(data) => {
+                    result.push(RequiredAction::WalletRpc(data.into()));
+                }
+                types::RequiredAction::Build(build) => {
+                    let resolved = self
```

Comment:
It looks like there is 2 sets of code to build/fetch the actions. Why is this?

Reply:
Not sure what you mean. Can you elaborate?